### DISCUSSION TITLE
How to include pylsp-mypy in the LSP config?

### DISCUSSION THREAD

**Discussion by ubaldot** [python-lsp/python-lsp-server#520] [Configuration]
I am using Vim9 [ LSP](https://github.com/yegappan/lsp) and I should tell it that I want to use `pylsp-mypy`, but I cannot understand how shall I do that. In my config I have the following, that I built according to [this](https://github.com/python-lsp/python-lsp-server/blob/develop/CONFIGURATION.md) guideline, but I don't know how to add `pylsp-mypy `

```
var pylsp_config = {
    'pylsp': {
        'plugins': {
            'pycodestyle': {
                'enabled': false},
            'pyflakes': {
                'enabled': true},
            'pydocstyle': {
                'enabled': false},
            'autopep8': {
                'enabled': false}, }, }, }
```

**Comment by smheidrich** [comment]
I use Neovim instead of Vim9 so I'm not entirely sure, but going by the format of the rest of your config, wouldn't it just be something like

```vim
var pylsp_config = {
  'pylsp': {
    'plugins': {
      'pylsp_mypy': {
        'enabled': true},
      ... }, }, }
```

?

Maybe this wasn't there when you originally asked the quesiton, but the [`pylsp-mypy` README](https://github.com/python-lsp/pylsp-mypy?tab=readme-ov-file#configuration) shows all config options prefixed with `pylsp.plugins.pylsp_mypy.`, the first two parts matching the "upper layers" of your existing config example. Hence why it seems correct to me.
